txnsync/incoming_test.go

```diff
@@ -0,0 +1,270 @@
+// Copyright (C) 2019-2021 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package txnsync
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/msgp/msgp"
+)
+
+type incomingLogger struct {
+	logging.Logger
+	lastLogged string
+}
+
+func (ml *incomingLogger) Debugf(format string, args ...interface{}) {
+	ml.lastLogged = fmt.Sprintf(format, args...)
+}
+
+func (ml *incomingLogger) Infof(format string, args ...interface{}) {
+	ml.lastLogged = fmt.Sprintf(format, args...)
+}
+
+func TestAsyncIncomingMessageHandlerAndErrors(t *testing.T) {
+
+	message := transactionBlockMessage{Version: 1}
+	messageBytes := message.MarshalMsg(nil)
+	sequenceNumber := uint64(1)
+	incLogger := incomingLogger{}
+
+	cfg := config.GetDefaultLocal()
+	s := syncState{
+		log: wrapLogger(&incLogger, &cfg)}
+
+	// expect UnmarshalMsg error
+	messageBytes[0] = 0
+	err := s.asyncIncomingMessageHandler(nil, nil, messageBytes, sequenceNumber)
+	msgpe := msgp.TypeError{}
+	require.True(t, errors.As(err, &msgpe))
+
+	// expect wrong version error
+	message = transactionBlockMessage{Version: 3}
```

Comment:
just for future compatibility, it would be better to use a version that we're not going to use.